MobileCenter/MobileCenter/MSMobileCenter.m

```diff
@@ -238,6 +238,9 @@ - (void)setEnabled:(BOOL)isEnabled {
   }
   self.enabledStateUpdating = NO;
   MSLogInfo([MSMobileCenter logTag], @"Mobile Center SDK %@.", isEnabled ? @"enabled" : @"disabled");
+
+  // Clean device history in case we are disabled.
+  [[[MSDeviceTracker alloc] init] clearDevices];
```

Comment:
This is executed whenever the state is going to be enabled or disabled but should only be executed is disabling. Maybe move this to applyPipelineEnabledState, the disabling/enabling is more obvious there.